controller.go

```diff
@@ -337,6 +341,14 @@ func (c *controller) clusterAgentInit() {
 			// service bindings
 			c.agentClose()
 			c.cleanupServiceBindings("")
+
+			c.Lock()
+			if c.agentStopDone != nil {
+				close(c.agentStopDone)
+				c.agentStopDone = nil
```

Comment:
If the channel is set to nil, it means the agent is not running at the moment the `AgentStopWait` was called. Then there won't be any wait.